tests/test_reader.py

```diff
@@ -423,6 +423,114 @@ def test_point():
         assert pt.mask == numpy.array([0])
         assert pt.band_names == ["b1"]
 
+        # buffer is useless with interpolate=False
+        with pytest.warns(UserWarning):
+            pt = reader.point(
+                src_dst,
+                [-57.566, 73.6885],
+                coord_crs="epsg:4326",
+                indexes=1,
+                nodata=1,
+                buffer=0.5,
+            )
+
+        # resampling_method is useless with interpolate=False
+        with pytest.warns(UserWarning):
+            pt = reader.point(
+                src_dst,
+                [-57.566, 73.6885],
+                coord_crs="epsg:4326",
+                indexes=1,
+                nodata=1,
+                resampling_method="bilinear",
+            )
+
+        # Interpolate=True but with Nearest, so no influence
+        pt = reader.point(
+            src_dst,
+            [-57.566, 73.6885],
+            coord_crs="epsg:4326",
+            indexes=1,
+            nodata=1,
+            interpolate=True,
+        )
+        assert pt.data == numpy.array([2800])
+        assert pt.mask == numpy.array([255])
+        assert pt.band_names == ["b1"]
+
+        # Interpolate=True + resampling=bilinear, default buffer
+        pt = reader.point(
+            src_dst,
+            [-57.566, 73.6885],
+            coord_crs="epsg:4326",
+            indexes=1,
+            nodata=1,
+            resampling_method="bilinear",
+            interpolate=True,
+        )
+        assert pt.data == numpy.array([2819])
```

Comment:
I think it makes sense to promote dtype from int->float for bilinear resampling:

```
gdallocationinfo -r bilinear  tests/fixtures/cog.tif -wgs84 -57.566 73.68856
Report:
  Location: (1090.59247446413P,1086.25414298276L)
  Band 1:
    Value: 2819.23468853823
```

Reply:
🤔 we currently don't expose `datatype` (`out_dtype`) 

I usually don't like when the API decide stuff for the user. We could add `out_dtype` option but I think it should be up to the user to ask for a specific output data type 

what do you think @scottyhq ?

Reply:
Yeah there are pros and cons for sure. I do think there should be an option though - consider asking for the  midpoint between values 0 & 1, I would want/expect 0.5 as a result.

Reply:
@scottyhq #796 will take care of this 

are we good to go with this PR?

Reply:
Sounds good. Though I think it'll be important to check that the resampling happens on promoted dtype, rather than interpolate on input dtype and finally cast to out_dtype.

Reply:
We'll check this when implementing the `out_dtype` option 🙏